discord/flags.py

```diff
@@ -0,0 +1,237 @@
+        flags via the properties rather than using this raw value.
+    """
+    __slots__ = ('value',)
+
+    def __init__(self, **kwargs):
+        self.value = self.ALL_OFF_VALUE
+        for key, value in kwargs.items():
+            if key not in self.VALID_FLAGS:
+                raise TypeError('%r is not a valid flag name.' % key)
+            setattr(self, key, value)
+
+    @classmethod
+    def _from_value(cls, value):
+        self = cls.__new__(cls)
+        self.value = value
+        return self
+
+    def __eq__(self, other):
+        return isinstance(other, SystemChannelFlags) and self.value == other.value
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __hash__(self):
+        return hash(self.value)
+
+    def __repr__(self):
+        return '<SystemChannelFlags value=%s>' % self.value
+
+    def __iter__(self):
+        for name, value in self.__class__.__dict__.items():
+            if isinstance(value, _flag_descriptor):
+                yield (name, self._has_flag(value.flag))
+
+    # For some reason the flags in the Discord API are "inverted"
+    # ergo, if they're set then it means "suppress" (off in the GUI toggle)
+    # Since this is counter-intuitive from an API perspective and annoying
+    # these will be inverted automatically
+
+    def _has_flag(self, o):
+        return (self.value & o) != o
+
+    def _set_flag(self, o, toggle):
+        if toggle is True:
+            self.value &= ~o
+        elif toggle is False:
+            self.value |= o
+        else:
+            raise TypeError('Value to set for SystemChannelFlags must be a bool.')
+
+    @_flag_descriptor
+    def join_notifications(self):
+        """:class:`bool`: Returns ``True`` if the system channel is used for member join notifications."""
+        return 1
+
+    @_flag_descriptor
+    def premium_subscriptions(self):
+        """:class:`bool`: Returns ``True`` if the system channel is used for Nitro boosting notifications."""
+        return 2
+
+
+@fill_with_flags
+class MessageFlags:
+    r"""Wraps up a Discord Message flag value.
+
+    See :class:`SystemChannelFlags`.
+
+    .. container:: operations
+
+        .. describe:: x == y
+
+            Checks if two flags are equal.
+        .. describe:: x != y
+
+            Checks if two flags are not equal.
+        .. describe:: hash(x)
+
+               Return the flag's hash.
+        .. describe:: iter(x)
+
+               Returns an iterator of ``(name, value)`` pairs. This allows it
+               to be, for example, constructed as a dict or a list of pairs.
+
+    Attributes
+    -----------
+    value: :class:`int`
+        The raw value. This value is a bit array field of a 53-bit integer
+        representing the currently available flags. You should query
+        flags via the properties rather than using this raw value.
+    """
+    __slots__ = ('value',)
+
+    def __init__(self, **kwargs):
+        self.value = self.ALL_OFF_VALUE
```

Comment:
The default flag value should be 0 since it's not inverted. That's the entire point of ALL_OFF_VALUE.

```suggestion
        self.value = 0
```